apps/omg_watcher/test/exit_processor/core_test.exs

```diff
@@ -766,6 +806,330 @@ defmodule OMG.Watcher.ExitProcessor.CoreTest do
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "fail when asked to produce proof for wrong badly encoded tx",
+         %{invalid_piggyback_on_input: %{state: state, request: request}, competing_transactions: [_, _, comp3 | _]} do
+      corrupted_txbytes = "corruption" <> Transaction.encode(comp3)
+      assert {:error, :malformed_transaction_rlp} = Core.get_input_challenge_data(request, state, corrupted_txbytes, 0)
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "fail when asked to produce proof for illegal oindex",
+         %{invalid_piggyback_on_input: %{state: state, request: request, ife_txbytes: txbytes}} do
+      assert {:error, :no_double_spent_on_particular_input} = Core.get_input_challenge_data(request, state, txbytes, -1)
+    end
+
+    @tag fixtures: [:alice, :bob, :processor_filled, :transactions, :in_flight_exits, :competing_transactions]
+    test "can produce the multiple challenge proofs for a input piggyback",
+         %{
+           alice: alice,
+           bob: bob,
+           processor_filled: state,
+           transactions: [tx | _],
+           competing_transactions: [comp1, comp2 | _],
+           in_flight_exits: [{ife_id, _} | _]
+         } do
+      txbytes = Transaction.encode(tx)
+      comp1_txbytes = Transaction.encode(comp1)
+      comp2_txbytes = Transaction.encode(comp2)
+
+      {:ok, recovered} = DevCrypto.sign(tx, [alice.priv, alice.priv]) |> Transaction.Recovered.recover_from()
+      %{sigs: comp1_signatures} = DevCrypto.sign(comp1, [bob.priv, alice.priv])
+      %{sigs: comp2_signatures} = DevCrypto.sign(comp2, [alice.priv, bob.priv])
+
+      comp1_ife_event = %{
+        call_data: %{in_flight_tx: comp1_txbytes, in_flight_tx_sigs: Enum.join(comp1_signatures)},
+        eth_height: 2
+      }
+
+      comp2_ife_event = %{
+        call_data: %{in_flight_tx: comp2_txbytes, in_flight_tx_sigs: Enum.join(comp2_signatures)},
+        eth_height: 2
+      }
+
+      comp1_ife_status = {1, <<1::192>>}
+      comp2_ife_status = {2, <<1::192>>}
+
+      {state, _} = Core.new_in_flight_exits(state, [comp1_ife_event], [comp1_ife_status])
+      {state, _} = Core.new_in_flight_exits(state, [comp2_ife_event], [comp2_ife_status])
+
+      {state, _} =
+        Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 0}, %{tx_hash: ife_id, output_index: 1}])
+
+      tx_blknum = 3000
+
+      {request, state} =
+        %ExitProcessor.Request{
+          blknum_now: 4000,
+          eth_height_now: 5,
+          piggybacked_blocks_result: [Block.hashed_txs_at([recovered], tx_blknum)]
+        }
+        |> Core.find_ifes_in_blocks(state)
+
+      assert {:ok, [%Event.InvalidPiggyback{txbytes: ^txbytes, inputs: [0], outputs: []}]} =
+               invalid_exits_filtered(request, state, only: [Event.InvalidPiggyback])
+
+      proofs =
```

Comment:
can we have this assertion more verbose? even if it makes it longer. If all you're interested in is the `decode!` being successful (I assume this is what you want in this test), then maybe just get `txbytes1, txbytes2 etc` from the pattern match and then `assert {:ok, ...} = Transaction.decode...`

Reply:
fixed